### PR TITLE
add phone to stg_zendesk__user

### PR DESCRIPTION
**Pull request by fivetran-jamie**
## PR Overview
**This PR will address the following Issue/Feature:**
https://github.com/fivetran/dbt_zendesk_source/issues/39

**This PR will result in the following new package version:**

v0.11.1 -- also this change won't be reflected in the transform package models

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Feature Updates
- Adds the `phone` field to `stg_zendesk__user`. 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [x] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
    
- [ ] docs were regenerated (unless this PR does not include any code or yml updates) -- WILL REGEN AFTER APPROVAL
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

see hex link in height

### If you had to summarize this PR in an emoji, which would it be?
 
☎️

### CODE REVIEW

File: models/stg_zendesk__user.sql [hunk 1]
```diff
@@ -37,6 +37,7 @@ final as (
         email,
         name,
         organization_id,
+        case when phone in ('-', 'null', 'NA') then null else phone end as phone,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_zendesk__user.sql:40]: Request to keep formatting of case when consistent.
```suggestion
        case when phone in ('-', 'null', 'NA') 
            then null 
            else phone 
        end as phone,
```

Comment (fivetran-joemarkiewicz) [reply models/stg_zendesk__user.sql:40]: Additionally, this seems like a real data quality issue on the connector side. Would you be able to follow up in the data team channel to understand why we sync the `phone` records like this? Also, are they any other `null` string types that we need to consider here? (ie. `NaN`, etc.)

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie this generally looks good but I just have one formatting suggestion and a general question for the Connector team before this will be good to approve.

Let me know if you have any questions.

File: models/stg_zendesk__user.sql (hunk 1 above)
Comment (fivetran-jamie) [reply models/stg_zendesk__user.sql]: Will do! And these were all the anomalous values that I saw in our internal Zendesk data at least, so it's unclear if other users would see different `null` string types.

Tried doing some digging from the Zendesk side... these [docs](https://support.zendesk.com/hc/en-us/articles/4408821302810-Understanding-how-Talk-calls-become-tickets#topic_hz3_5nm_sx) seem to suggest that the `phone` field is automatically populated by the system, but that internal users can also manually add phone numbers to profiles 🤔

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM!

**Review by fivetran-catfritz — APPROVED** [review]
approved for release!

**Pull request merged**